### PR TITLE
Div 8591 | KYC Flow in Public App

### PR DESCRIPTION
**Pull request by varadeth**

### CODE REVIEW

File: public_app/src/components/Registration/AddMember/FormPersonalDetails.js [hunk 1]
```diff
@@ -179,13 +164,49 @@ export const FormPersonalDetails = ({ setValue, formData, navigation, verifyDeta
...
             });
     };
+
+    const updateFormData = (kycData) => {
+        let selectedDistrict = null;
+        stateAndDistricts.states.forEach(state => {
+            if(state.name === kycData.state) {
+                state.districts.forEach(district => {
+                    if(kycData.district !== "" && district.name.indexOf(kycData.district) >= 0) {
+                        selectedDistrict = district.name;
```
Comment (egov-joy) [inline public_app/src/components/Registration/AddMember/FormPersonalDetails.js:174]: could we please do a devbox on this? We should be able to do this in a simpler manner

Comment (varadeth) [reply public_app/src/components/Registration/AddMember/FormPersonalDetails.js:174]: it needs two loops to get to correct district
at best what we could do is refactor this to look better

File: public_app/src/components/CitizenLogin/citizen-login.js [hunk 2]
```diff
@@ -49,20 +55,28 @@ export function CitizenLoginComponent(props) {
...
                 .then((response) => {
-                    setState((prevState) => {
-                        return {
-                            ...prevState,
-                            showOnlyOTP: !prevState.showOnlyOTP,
-                            invalidOTP: "",
-                            invalidMobileNumber: ""
-                        }
-                    })
+                    if(response.status === 200) {
```
[the remark this answers is not in the document]
Comment (varadeth) [reply public_app/src/components/CitizenLogin/citizen-login.js]: this is removed

File: public_app/src/components/CitizenLogin/citizen-login.js [hunk 3]
```diff
@@ -49,20 +55,28 @@ export function CitizenLoginComponent(props) {
...
+                        });
+                    }
                 }).catch((error) => {
                 console.log(error)
                 alert(error)
             })
         }
     };
+
+    const verifyOTPHandler = () => {
```
[the remark this answers is not in the document]
Comment (varadeth) [reply public_app/src/components/CitizenLogin/citizen-login.js]: this is removed

File: public_app/src/components/CitizenLogin/citizen-login.js [hunk 4]
```diff
@@ -102,14 +116,26 @@ export function CitizenLoginComponent(props) {
...
                                className="form-control form-control-lg"
                                onChange={setMobileNumber}
                                value={state.phoneNumber}
                                disabled={!state.showOnlyOTP}
                                maxLength={10}
-                        />
+                            />
+                        }
+                        {state.individualId &&
+                            <input placeholder={t('login.mobPlaceholder')}
```
[the remark this answers is not in the document]
Comment (varadeth) [reply public_app/src/components/CitizenLogin/citizen-login.js]: this is removed

File: public_app/public/locales/en/translation.json [hunk 5]
```diff
@@ -123,7 +123,13 @@
...
       "emailPlaceholder": "Enter Email ID",
       "verifyEmailLabel": "Verify Beneficiary Email ID",
-      "confirmEmailPlaceholder": "Confirm Email ID"
+      "confirmEmailPlaceholder": "Confirm Email ID",
+      "mosipEnroll": "Enroll using MOSIP ID",
+      "mosipTitle": "Verify with Mosip",
+      "mosipIndividualIdType": "Select Individual Type",
+      "mosipVid": "Enter VID",
+      "mosipUin": "Enter UIN",
+      "mosipKycConsent": "I hereby have taken consent of the beneficiary to use their KYC information for the purpose of registering in the vaccination program."
```
[the remark this answers is not in the document]
Comment (varadeth) [reply public_app/public/locales/en/translation.json:132]: will change this message

**Pull request merged**